stdlib/public/core/Bool.swift

```diff
@@ -86,6 +86,28 @@ public struct Bool {
   public init(_ value: Bool) {
     self = value
   }
+
+  /// Returns a random Boolean value
+  ///
+  /// - Parameter generator: The random number generator to use when getting a
+  ///   random Boolean.
+  /// - Returns: A random Boolean value.
+  @inlinable
+  public static func random<T: RandomNumberGenerator>(
+    using generator: inout T
+  ) -> Bool {
+    return (generator.next() >> 17) & 1 == 0
```

Comment:
This isn't totally insane, because for very weak RNGs middle bits have better randomness properties than low bits, but really no one should be using those ...

Reply:
Nate sent this in the initial pr:
> For some random generators the low bits have higher levels of repetition than the higher bits do. Could we change this to (generator.next() >> 17) & 1 == 0?

Reply:
Got it; seemed strange because `17` is pretty arbitrary. If we want to use high bits, `Int(bitPattern: generator.next()) < 0` would be less mystifying, IMO.

Reply:
@xwu you don't want the *high* bits either, those can also be weak. You really want some "middle" bits. This is worth a comment, though.

Reply:
Go ahead, try to name a number more random than `17`.

Reply:
FWIW, according to some cursory testing, the `GKRandomSource` types use `31` for the offset in `nextBool()` (i.e., the high bit, like @xwu suggested).